### DISCUSSION TITLE
Idea: isolate template / photos from website generation

### DISCUSSION THREAD

**Discussion by nicosomb** [waschinski/photo-stream#92] [Ideas]
Thank you for your tool. 

Website generation should be isolated from website content (layout and pictures). 
That would improve the usage of photo-stream. 

With this change, we can imagine to have only a repository with the layout, an other one with pictures. 

And a script (or a github action) will build the website content by using photo-stream and the 2 other repositories. 

I don't know if it's easy to imagine and I don't know if I can help you.

**Comment by kwickramasekara** [comment]
Hi @nicosomb my [pet project](https://github.com/kwickramasekara/photo-grid) might be what you need. Not trying to poach folks from this project, but I know this is a significant shift from the current implementation. 

I'v been a user of Photo Stream for over 5 years but TBH, I don't know Ruby well enough to contribute to the project. So I went ahead and created a similar project with a de-coupled architecture keeping the front-end site and photos separate. It uses a headless-CMS for storing photos, so that comes with a lot of benefits like on-the-fly image resizes, loading photos from a CDN for better performance and etc. You can see the full [list of features](https://github.com/kwickramasekara/photo-grid/wiki/Features) including a backend dashboard that allows you to upload and manage photos easily. A [demo site](https://photo-grid-demo.netlify.app/) is also available to preview.

If you have any questions, please use our [Q&A](https://github.com/kwickramasekara/photo-grid/discussions/categories/q-a) section and I'd be happy to help!

P.S: the project shows as "Work in Progress" currently, but it should be stable to use. I am just finishing up documentation at the moment.

**Comment by nicosomb** [comment]
@kwickramasekara wow nice! I will have a look.

**Comment by waschinski** [comment]
@kwickramasekara thanks for sharing your project which btw looks awesome!

To be honest, I have been considering a complete rewrite of Photo Stream over trying to squeeze more features/changes into the current code base. I would also not use Ruby. 😅 But since I am currently not actively using Photo Stream myself any more and free time is limited, it's hard to justify spending a lot of time on this.

Also to be fair, Photo Stream has initially been created by a designer more than a programmer afaik. It had a simple approach in mind and does this well. The long processing times which have been reported in the last couple of months are something I still want to look at but I am also not very experienced with Ruby so there's that.

That being said, @nicosomb I would not keep your hopes up on fundamental changes to Photo Stream like you have proposed and consider photo-grid as a more feature rich alternative.

**Comment by kwickramasekara** [comment]
@waschinski yea, I am actually a user since the days when Tim maintained the project. I love the minimal approach of Photo Stream and thats why Photo Grid also looks very similar.

Thank you for taking over and keeping Photo Stream alive - I know its not easy maintaining an open source project 🙏
